pkg/controller/history/controller_history.go

```diff
@@ -46,10 +46,10 @@ import (
 const ControllerRevisionHashLabel = "controller.kubernetes.io/hash"
 
 // ControllerRevisionName returns the Name for a ControllerRevision in the form prefix-hash. If the length
-// of prefix is greater than 223 bytes, it is truncated to allow for a name that is no larger than 253 bytes.
+// of prefix is greater than 52 bytes, it is truncated to allow for a name that is no larger than 63 bytes.
 func ControllerRevisionName(prefix string, hash uint32) string {
-	if len(prefix) > 223 {
-		prefix = prefix[:223]
+	if len(prefix) > 52 {
+		prefix = prefix[:52]
```

Comment:
is this going to change the computed name of existing controllers' revisions and create duplicates when someone upgrades to 1.11?

Reply:
@liggitt `func NewControllerRevision` is only used by Statefuset. Pod creation error would occur if len(controllerRevision.Name)>63 (I doubt if any users wolud take names that long). It behaves the same as before if the length of existing controllers' revisions names (`prefix-hash`) are all less than or equal 63. I don't know if I get it right.